modules/components/AsyncConnect.js

```diff
@@ -53,16 +55,14 @@ export class AsyncConnect extends Component {
   }
 
   componentWillReceiveProps(nextProps) {
+    const navigated = this.props.location !== nextProps.location;
```

Comment:
if `location` is a router object, you should do a deepDiff to compare objects.

Reply:
as far as I'm aware object is changing every time a route changes, so its enough to do a shallow check as pointers will always be different unless it is the same location